Composer/packages/server/src/models/connector/csharpBotConnector.ts

```diff
@@ -1,92 +1,205 @@
+      });
+      buildDebug('building bot runtime: %d', build.pid);
+      build.stdout &&
+        build.stdout.on('data', function(str) {
+          buildDebug('%s', str);
+        });
+      build.stderr &&
+        build.stderr.on('data', function(err) {
+          reject(err.toString());
+        });
+      build.on('exit', function(code) {
+        resolve(code);
+      });
+    });
   };
 
-  sync = async (config: DialogSetting) => {
-    // archive the project
-    // send to bot runtime service
+  private getConnectorConfig = (config: DialogSetting) => {
+    const configList: string[] = [];
+    if (config.MicrosoftAppPassword) {
+      configList.push('--MicrosoftAppPassword');
+      configList.push(config.MicrosoftAppPassword);
+    }
+    if (config.luis) {
+      if (config.luis.authoringKey) {
+        configList.push('--luis:endpointKey');
+        configList.push(config.luis.authoringKey);
+      }
+      if (config.luis.authoringRegion) {
+        configList.push('--luis:endpoint');
+        configList.push(`https://${config.luis.authoringRegion}.api.cognitive.microsoft.com`);
+      }
+    }
+
+    return configList;
+  };
+
+  private addListeners = (child: ChildProcess, handler: Function, resolve: Function, reject: Function) => {
+    const currentDebugger = runtimeDebugs[child.pid];
+    let erroutput = '';
+    child.stdout &&
+      child.stdout.on('data', (data: any) => {
+        currentDebugger('bot runtime (%d): %s', child.pid, data);
+        resolve(child.pid);
+      });
+
+    child.stderr &&
+      child.stderr.on('data', (err: any) => {
+        erroutput += err.toString();
+      });
+
+    child.on('exit', code => {
+      currentDebugger('exit %d', code);
+      handler();
+      if (code !== 0) {
+        currentDebugger('exit %d: %s', code, erroutput);
+        reject(erroutput);
+      }
+    });
+
+    child.on('message', msg => {
+      currentDebugger('bot runtime received: %s', msg);
+    });
+  };
+
+  private getBotPath = () => {
     const currentProject = BotProjectService.getCurrentBotProject();
     if (currentProject === undefined) {
       throw new Error('no project is opened, nothing to sync');
     }
-    const dir = Path.join(currentProject.dataDir);
-    const luisConfig = currentProject.luPublisher.getLuisConfig();
-    await this.archiveDirectory(dir, './tmp.zip');
-    const content = fs.readFileSync('./tmp.zip');
+    return Path.join(currentProject.dir);
+  };
 
-    const form = new FormData();
-    form.append('file', content, 'bot.zip');
+  private start = async (dir: string, config: DialogSetting): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      const runtime = spawn(
+        'dotnet',
+        [
+          'bin/Debug/netcoreapp2.1/BotProject.dll',
+          `--urls`,
+          this.endpoint,
+          ...this.getConnectorConfig(config),
+          '--environment',
+          'development',
+        ],
+        {
+          detached: true,
```

Comment:
Why do we want this to be detached? It seems reasonable that this node process can be made a parent of the dotnet one?

Reply:
removed.